src/test/scala/chisel3/tests/QueueTest.scala

```diff
@@ -57,4 +59,28 @@ class QueueTest extends FlatSpec with ChiselScalatestTester {
       }.join()
     }
   }
+
+  it should "work with IrrevocableIO" in{
+    test(new Module{
+      val io = IO(new Bundle{
+        val in = Flipped(Irrevocable(UInt(8.W)))
+        val out = Irrevocable(UInt(8.W))
+      })
+      val full = RegInit(false.B)
+      when(io.in.fire()){ full := true.B }.elsewhen(io.out.fire()){ full := false.B }
+      io.in.ready := !full
+      io.out.valid := full
+      io.out.bits := RegNext(io.in.bits)
+    }){c =>
+      c.io.in.initSource().setSourceClock(c.clock)
+      c.io.out.initSink().setSinkClock(c.clock)
+      c.io.in.enqueue(9.U)
+      c.io.out.expectDequeue((9.U))
+      parallel(
+        c.io.in.enqueueSeq(Seq(5.U, 2.U)),
```

Comment:
Would it make sense to add the 9.U above to the enqueueSeq? Or is the above meant to specifically test cycle-by-cycle behavior?

Reply:
It is just some redundant cycly-by-cycle test. Removed in 49d2857